Change input from word id til literal in requestWord

Trudellik opened the pull request.

Trudellik committed:
WordId is replaced to literal in requesttoggle

Trudellik committed:
Debug errors

Trudellik committed:
Finalize the missing part of converting

Trudellik committed:
StyleCI formatting

Thanerik commented on app/Http/Controllers/RequestController.php:
```diff
@@ -63,18 +65,15 @@ public function getList(int $limit, ?SortInput $sort)
      * @return Word
      */
-    public function toggleRequest($wordId, $user): Word
+    public function toggleRequest($literal): Word
```
Her kunne jeg faktisk helt vildt ønske mig at du lave en unittest til - hvor du tester alle veje:
1. Toogle en eksisterende ord med ingen request op
2. Toogle en eksisterende ord med 1 requests, både op og ned
3. Toogle en ukendt ord med ingen request
4. Toogle en eksisterende ord med ingen request NED (Slet ikke muligt, men hvis... hvad så?)
5. En scenarie hvor delete af ord faktisk fejler...

Thanerik commented on app/Http/Controllers/TranslationController.php:
```diff
@@ -26,8 +30,9 @@ public function __construct(WordService $wordService, SignService $signService,
     public function createTranslation(Request $request)
     {
         $user = $this->userService->getUser();
+        $literal = $request->input('literal');
```
Der er kun en `$args['literal']` fra endpoint. Der er ingen `$request->input('literal')`... Medmindre du mimer min sortInput og laver en FormRequest fra bunden med form validation - så kan du lave en `$request = new WordRequest($args['literal'])` fra endpoint og give det videre hertil...

Thanerik replied:
Altså: `php artisan make:request WordRequest` - vupti!

Trudellik replied:
For at undgå scope creep, opretter jeg en issue som jeg vil arbejde videre på det.

Thanerik commented on app/Http/Controllers/TranslationController.php:
```diff
@@ -47,6 +52,8 @@ public function createTranslation(Request $request)
 
             return response()->json($translation);
         } else {
+            \Log::info('Creation of translation failed - how can it happens?', [$word, $sign, $desc]);
```
Hvorfor har du ikke taget min råd med dig, hvor du i sidste PR har rettet det til?

Thanerik replied:
Ah, det er ikke merget med i denne branch... Kan du gøre det?

Trudellik replied:
Lige præcist, jeg har ikke merget denne branch fra dev indtil nu.

Thanerik commented on database/seeds/DatabaseSeeder.php:
```diff
@@ -40,7 +40,7 @@ private function output(string $name, Model $model)
         //$visible = $model::all()->count();
         //$trashed = $model::onlyTrashed()->count();
-        $total = $model::withTrashed()->count();
+        $total = $model::count();
```
Og her vil du opnå...?

Trudellik replied:
Vi er blevet enige om at droppe softdelete i Word, hvorfor withTrashed ikke vil virke.

Thanerik replied:
Problemet her er at det ikke kun er til Word modellen - men en generelt metode til alle modeller vi har seedet ind - og nogle af dem har softdelete sat til - så vil tallene ser lidt mærkelig ud

Trudellik replied:
Hentyder du også sign og description? Jeg er ikke helt med hvad du mener.

Thanerik replied:
![image](https://user-images.githubusercontent.com/6329700/71602444-bad72600-2b58-11ea-8129-2303bbb9a2ba.png)
Nej, det bliver også brugt på Signs, Descriptions, Users og Translations...

Thanerik replied:
Så hvis du kan lave en mere generelt fiks, der tager højde på softdeletes, og ikke-softdeletes, er det bedre

Trudellik replied:
En løsning er implementeret

Thanerik replied:
Godkendt

Thanerik commented on resources/lang/da/error.php:
```diff
@@ -0,0 +1,5 @@
+<?php
+
+return [
+    'toggle.request.failed'    => 'Opdatering af efterspørgslen mislykkedes',
```
Der er vist flere rækker her end den ene?

Trudellik replied:
Der vil komme flere løbende :) Indtil videre er det kun 3.

Du kan bare oprette en issue, hvis du har fundet et sted hvor der mangler at smide en fejlbesked til fronten.

Thanerik replied:
Næ, det er blot fordi di ikke har merget dev ind, da jeg mente at error.php har flere rækker fejlebeskeder som du har tilføjet.

Thanerik commented on app/Repositories/WordRepository.php:
```diff
@@ -89,8 +89,8 @@ public function firstOrNew(string $literal, Language $language, $user): Word
+    public function getWordWithRequesters(Word $word): Word
     {
-        return Word::withCount('requesters')->find($wordId);
+        return Word::withCount('requesters')->find($word->id)->first();
```
Hvad er forskellen?
Og har du taget højde for at den kan komme lige efter man har nedvotet en word til nul request, og word'en dermed slettes? Denne metode er 100% sikker på at altid returerer en Word. Det er så ikke muligt at returner en null.

Trudellik replied:
Det kommer an på om vil du modtage en eloquent eller en Word-model. Jeg fjerner bare den.

Thanerik replied:
Bare en string... en id. Det er også meget fint i denne tilfælde. Så kan de andre steder også kalde det med id alene, og det vil så virke meget mere generelt end at fastelåse os til at skal have word. Så skal jeg lave `$word = Word::find($wordId)` før jeg kan kalde `WordRepository->($word)` - hvis jeg kun har en $wordId at råde over.

Trudellik replied:
Metoden bliver kun brugt ifm toggleRequest, som formålet med denne issue at ændre fra wordId til literal. Så jeg forstår ikke hvorfor du stadig forventer at API'en spytter stadig med wordId?

Thanerik replied:
Jo - problemet er blot at du pt. er ind i `WordRepository` - som er forventet at skulle kunne håndtere på tværs af en masse af services og controllers. I lige netop det tilfælde med toogleRequest, har vi allerede en Word model som vi kan anvende til denne metode. MEN der kunne jo være i fremtiden at der dukker en anden behov op, der også har brug for at fetche en word med requesters - og hvis den anden metode kun har en wordId at råde over, så ville det være dumt at omgøre det hele, når vi i denne metode allerede kan "begrænse" os til at kun modtage WordId, da det er den eneste variable vi har brug for, før at løse lige netop den opgave som metoden gør... Metoden har ikke brug for en word model - blot en id over Word...

Trudellik committed:
merge + rettelse efter review

Trudellik committed:
fix echo of softdelete and non-softdelete

Trudellik committed:
Merge branch 'develop' into feature/debug_requestWordAPI

Thanerik commented:
This fixes #33

Trudellik committed:
Merge branch 'develop' into feature/debug_requestWordAPI

Trudellik committed:
Setup of test

- test with graphql query
- DB migration for test

Trudellik committed:
Adjustment in API and migration

- @auth is removed, because it will not works with @field
- dropForeign does not support in sqlLite, so sqlType check is added in down
- Setup of test updated

Trudellik committed:
Merge branch 'develop' into feature/debug_requestWordAPI

Trudellik committed:
Fix test run of phpunit

Trudellik committed:
Merge branch 'develop' into feature/debug_requestWordAPI

Trudellik committed:
Minor steps forward

Thanerik commented:
Du siger bare til når du gerne vil have at jeg skal reviewe det

Trudellik committed:
Merge branch 'develop' into feature/debug_requestWordAPI

Trudellik committed:
Revert "Setup of test"

This reverts commit 787d872dc7dac8e17bbf49c61696f3dedd1f7e3a.

Trudellik committed:
Merge branch 'develop' into feature/debug_requestWordAPI

Trudellik committed:
Foreign keys file reverted

Trudellik committed:
Merge remote-tracking branch 'origin/feature/debug_requestWordAPI' into feature/debug_requestWordAPI

Trudellik committed:
StyleCI commit
